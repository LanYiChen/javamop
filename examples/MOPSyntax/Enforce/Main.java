class SimpleRunnable1 implements Runnable{
    public void run(){
        System.out.println("Thread " + Thread.currentThread().getName());
    }
}

class SimpleRunnable2 implements Runnable{
    public void run(){
        System.out.println("Thread " + Thread.currentThread().getName());
    }
}

public class Main{
    public static void main(String args[]) throws Exception{
        Thread t1 = new Thread (new SimpleRunnable1());
        Thread t2 = new Thread (new SimpleRunnable2());
        t1.setName("T1");
        t2.setName("T2");
        t1.start();
        t2.start();
        t1.join();
        t2.join();
    }
}
