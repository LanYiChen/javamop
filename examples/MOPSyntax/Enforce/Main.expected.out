Thread Thread-1
Thread Thread-0
success!
